### PR TITLE
Django1.6

### PR DESCRIPTION
**Pull request by snopoke**
- https://github.com/dimagi/dimagi-utils/pull/247
- https://github.com/dimagi/couchlog/pull/17

### CODE REVIEW

File: corehq/apps/api/resources/v0_5.py [hunk 1]
```diff
@@ -250,7 +250,7 @@ def patch_list(self, request=None, **kwargs):
         (BSD licensed) and modified to pass the kwargs to `obj_create` and support only create method
         """
         request = convert_post_to_patch(request)
-        deserialized = self.deserialize(request, request.raw_post_data, format=request.META.get('CONTENT_TYPE', 'application/json'))
+        deserialized = self.deserialize(request, request.body, format=request.META.get('CONTENT_TYPE', 'application/json'))
```
Comment (dimagimon) [inline corehq/apps/api/resources/v0_5.py]: E501 line too long (123 > 115 characters)

File: corehq/apps/api/resources/v0_5.py [hunk 2]
```diff
@@ -281,7 +281,7 @@ def post_list(self, request, **kwargs):
         Exactly copied from https://github.com/toastdriven/django-tastypie/blob/v0.9.14/tastypie/resources.py#L1314
         (BSD licensed) and modified to catch Exception and not returning traceback
         """
-        deserialized = self.deserialize(request, request.raw_post_data, format=request.META.get('CONTENT_TYPE', 'application/json'))
+        deserialized = self.deserialize(request, request.body, format=request.META.get('CONTENT_TYPE', 'application/json'))
```
Comment (dimagimon) [inline corehq/apps/api/resources/v0_5.py]: E501 line too long (123 > 115 characters)

**Comment by czue** [comment]
build fails looks likely real:

```
======================================================================
FAIL: test_update (corehq.apps.api.tests.TestCommCareUserResource)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/api/tests.py", line 423, in test_update
    self.assertEqual(response.status_code, 202, response.content)
AssertionError: 200 != 202 : {"default_phone_number": "+50253311399", "email": "tlast@example.org", "first_name": "test", "groups": ["bdc1004b2a4e8ba605289034959a19a1"], "id": "bdc1004b2a4e8ba605289034954d618c", "language": "pol", "last_name": "last", "phone_numbers": ["+50253311399", "50253314588"], "resource_uri": "/a/qwerty/api/v0.5/user/bdc1004b2a4e8ba605289034954d618c/", "user_data": {"chw_id": "13/43/DFA"}, "username": "test"}
----------------------------------------------------------------------
Ran 85 tests in 539.934s
```

File: corehq/apps/api/resources/v0_5.py (hunk 1 above)
Comment (dimagimon) [inline corehq/apps/api/resources/v0_5.py]: E501 line too long (123 > 115 characters)

File: corehq/apps/api/resources/v0_5.py (hunk 2 above)
Comment (dimagimon) [inline corehq/apps/api/resources/v0_5.py]: E501 line too long (123 > 115 characters)

File: corehq/apps/api/tests.py [hunk 3]
```diff
@@ -420,7 +420,7 @@ def test_update(self):
         response = self.client.put(self.single_endpoint(backend_id),
                                    simplejson.dumps(user_json),
                                    content_type='application/json')
-        self.assertEqual(response.status_code, 202, response.content)
+        self.assertEqual(response.status_code, 200, response.content)
```
Comment (snopoke) [inline corehq/apps/api/tests.py]: Changed in Tastypie v0.9.16: https://github.com/django-tastypie/django-tastypie/commit/abc0bef

File: corehq/apps/api/resources/v0_5.py (hunk 2 above)
Comment (dimagimon) [inline corehq/apps/api/resources/v0_5.py]: E501 line too long (123 > 115 characters)

**Comment by millerdev** [comment]
Is it too big a jump to go right to 1.7?

**Comment by snopoke** [comment]
In my opinion yes but feel free to make up your own mind:
https://docs.djangoproject.com/en/1.7/releases/1.7/

**Comment by millerdev** [comment]
My question should have been "Are these changes also forward-compatible with Django 1.7?"

**Comment by snopoke** [comment]
Oh, yes they are.

**Comment by millerdev** [comment]
:+1:

File: corehq/apps/api/resources/v0_5.py (hunk 1 above)
Comment (dimagimon) [inline corehq/apps/api/resources/v0_5.py]: E501 line too long (123 > 115 characters)

**Comment by twymer** [comment]
:tada:

**Comment by czue** [comment]
@snopoke I think we should merge some of the backwards compatible changes in now (will allow us to do soft QA locally on django 1.6 without having to use this branch)

**Comment by czue** [comment]
(actually i see the django version bump isn't in requirements so maybe that's what you're proposing here, just waiting for quick QA?)

**Comment by snopoke** [comment]
Yea I removed the 1.6 version bump. This is all backwards compatible as far
add I can tell.

One thing outstanding before we can test on staging is we need to bump the
celery version to >3.1.

**Comment by snopoke** [comment]
Kicked build

**Comment by snopoke** [comment]
@czue :white_check_mark:

**Comment by czue** [comment]
nice. i see the QA label is still on. are you wanting to test the tastypie upgrade or fine for this to get merged?

**Comment by snopoke** [comment]
Fine to merge

**Comment by twymer** [comment]
:clap:

**Pull request merged**